### ISSUE TITLE
[jit] set up views for Autodiff and autograd hooks 

### ISSUE THREAD

**Issue by wanchaol** [pytorch/pytorch#20734]
Derived from bug #19769 , Here is what we found that is missing in our Autodiff infrastructure.  In Autodiff, we don't set up the views for inputs/outputs after we execute the forward graph(see this [TODO](https://github.com/pytorch/pytorch/blob/be33434d85953fe6f237d475cc8ae2e9a822ed34/torch/csrc/jit/graph_executor.cpp#L350)). This disables us from adding complicated Ad formulas (such as `linear` and `layer_norm`, etc.), because when we calculate higher level gradients, we usually ended up capturing multiple outputs, and some of the output is just a view of another output, and we want the gradient to flow to the original output, which we could not do in the current autodiff infrastructure.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
